### PR TITLE
Bug fix/update foxx services after restore

### PR DESCRIPTION
**Pull request by hkernbach**
This makes Foxx services publicly available after a successful restore of the "_apps" collection.
This will only work if foxx-queues are **NOT** disabled. 

- restore tests needs to be added - 

If the review is fine, PR's for additional versions are needed.

### CODE REVIEW

File: arangod/RestHandler/RestImportHandler.cpp [hunk 1]
```diff
@@ -1000,6 +1000,13 @@ Result RestImportHandler::performImport(SingleCollectionTransaction& trx,
     res = opResult.result;
   }
 
+  if (collectionName == "_appbundles") {
```
Comment (mchacki) [inline arangod/RestHandler/RestImportHandler.cpp]: Why is it _appbundles here? and _apps below?

I think we need to trigger it if either one is imported:
```suggestion
  if (collectionName == "_appbundles" || collectionName == "_apps") {
```

File: arangod/RestHandler/RestReplicationHandler.cpp [hunk 2]
```diff
@@ -842,6 +842,13 @@ void RestReplicationHandler::handleCommandRestoreData() {
                         res.errorMessage());
     }
   } else {
+    // if result is ok and the collection is the '_apps' service collection,
+    // we need to trigger the foxx selfHeal function in order to make them
+    // publicly available.
+    if (colName == "_apps") {
```
Comment (mchacki) [inline arangod/RestHandler/RestReplicationHandler.cpp]: See above.
```suggestion
  if (collectionName == "_appbundles" || collectionName == "_apps") {
```

File: arangod/RestHandler/RestImportHandler.cpp (hunk 1 above)
Comment (hkernbach) [reply arangod/RestHandler/RestImportHandler.cpp]: Yes, we could do it like this. But we should discuss further. If that is the right way to do it.

File: arangod/RestHandler/RestReplicationHandler.cpp (hunk 2 above)
Comment (hkernbach) [reply arangod/RestHandler/RestReplicationHandler.cpp]: Yes, we could do it like this. But we should discuss further. If that is the right way to do it.

**Comment by hkernbach** [comment]
> I think this is not completely consistent.
Also i am missing CHANGELOG entry.

Yes, this is still WIP.

**Comment by mchacki** [comment]
New Jenkins run:

http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2199/

New implementation of the fix.

**Comment by mchacki** [comment]
@Simran-B FYI: This newest fix does not rely on foxx queues to be present.

**Pull request merged**